src/transport-layer.service.ts

```diff
@@ -0,0 +1,58 @@
+import axios, { AxiosRequestConfig, AxiosInstance } from 'axios';
+import { IServerConfiguration } from './shared/interfaces';
+import ITransportLayer from './transport.interface';
+
+const config: IServerConfiguration = {
+    baseURL: 'http://127.0.0.1:3337',
+    timeout: 100000,
+    headers: {},
+};
+
+export default class TransportLayer implements ITransportLayer {
+    private instance: AxiosInstance;
+
+    constructor() {
+        this.instance = axios.create(config);
```

Comment:
You can add interceptors to axios globally. You can import axios object from library and add interceptors to this object. Benefits: you can make `TransportLayer` methods static and avoid `TransportLayer` instances creating in your components. Also you can implement Singleton pattern for this service